experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -131,3 +148,98 @@ export interface List<TTypes extends AllowedTypes> extends ReadonlyArray<UnboxNo
 	// 	source: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
 	// ): void;
 }
+
+/**
+ * An object which supports property-based access to fields.
+ * @alpha
+ */
+export type SharedTreeObject<TSchema extends StructSchema> = ObjectFields<
+	TSchema["structFieldsObject"]
+>;
+
+/**
+ * Helper for generating the properties of a {@link SharedTreeObject}.
+ * @alpha
+ */
+export type ObjectFields<TFields extends RestrictiveReadonlyRecord<string, FieldSchema>> = {
+	// Add getter only (make property readonly) when the field is **not** of a kind that has a logical set operation.
```

Comment:
Is this still true given the new typing?

Reply:
Right now this means we won't allow the setting of lists (and of node identifiers). I actually can't remember right now whether we said we can set lists or not. `myNode.list = [a, b, c]`, as opposed to having to delete all elements and then add new elements. I think we don't allow the setting, but I could have forgotten. Do you remember?

Reply:
right, this requires slice delete. nvm.